nav2_costmap_2d/plugins/range_sensor_layer.cpp

```diff
@@ -0,0 +1,525 @@
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <angles/angles.h>
+#include <algorithm>
+#include <list>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "pluginlib/class_list_macros.hpp"
+#include "geometry_msgs/msg/point_stamped.hpp"
+#include "nav2_costmap_2d/range_sensor_layer.hpp"
+
+PLUGINLIB_EXPORT_CLASS(nav2_costmap_2d::RangeSensorLayer, nav2_costmap_2d::Layer)
+
+using nav2_costmap_2d::LETHAL_OBSTACLE;
+using nav2_costmap_2d::INSCRIBED_INFLATED_OBSTACLE;
+using nav2_costmap_2d::NO_INFORMATION;
+
+using namespace std::literals::chrono_literals;
+
+namespace nav2_costmap_2d
+{
+
+RangeSensorLayer::RangeSensorLayer() {}
+
+void RangeSensorLayer::onInitialize()
+{
+  current_ = true;
+  buffered_readings_ = 0;
+  last_reading_time_ = node_->now();
+  default_value_ = to_cost(0.5);
+
+  matchSize();
+  resetRange();
+
+  declareParameter("enabled", rclcpp::ParameterValue(true));
+  node_->get_parameter(name_ + "." + "enabled", enabled_);
+  declareParameter("phi", rclcpp::ParameterValue(1.2));
+  node_->get_parameter(name_ + "." + "phi", phi_v_);
+  declareParameter("inflate_cone", rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + "." + "phi", phi_v_);
+  declareParameter("no_readings_timeout", rclcpp::ParameterValue(0.0));
+  node_->get_parameter(name_ + "." + "no_readings_timeout", no_readings_timeout_);
+  declareParameter("clear_threshold", rclcpp::ParameterValue(0.2));
+  node_->get_parameter(name_ + "." + "clear_threshold", clear_threshold_);
+  declareParameter("mark_threshold", rclcpp::ParameterValue(0.8));
+  node_->get_parameter(name_ + "." + "mark_threshold", mark_threshold_);
+  declareParameter("clear_on_max_reading", rclcpp::ParameterValue(false));
+  node_->get_parameter(name_ + "." + "clear_on_max_reading", clear_on_max_reading_);
+
+  double temp_tf_tol = 0.0;
+  declareParameter("transform_tolerance", rclcpp::ParameterValue(0.0));
```

Comment:
```suggestion
  declareParameter("transform_tolerance", rclcpp::ParameterValue(0.2));
```

Reply:
This is also a new parameter in the name_ namespace because you used the `declareParameter` override, that's not what `get_parameter` is getting so these are misaligned. This is getting the costmap's `transform_tolerance` value **which is totally fine** but then don't declare it incorrectly. 

If you choose to have another transform tolerance for the range layer separate of the costmap_2d one, you need to add it to the markdown parameter file and navigation2 website, its a new parameter.